tests/pip_deps/pip_deps.bzl

```diff
@@ -0,0 +1,55 @@
+""" A demo implementation for pip_deps which provides @unpinned_pip//:pin. """
+
+load("@rules_python//python:pip.bzl", "pip_parse")
+
+def _requirements_in_impl(repository_ctx):
+    repository_ctx.file(
+        "requirements.in",
+        content = "".join(["{package} == {version}\n".format(
+            package = package,
+            version = version,
+        ) for (package, version) in repository_ctx.attr.packages.items()]),
+    )
+    repository_ctx.file("WORKSPACE", content = "")
+    repository_ctx.file("BUILD", content = """
+load("@rules_python//python:pip.bzl", "compile_pip_requirements")
+
+compile_pip_requirements(
```

Comment:
this is really clever to drop the updater binary into an external repo like this :)

Reply:
Haha, thanks!